core/tests/protractor/stateEditor.js

```diff
@@ -389,7 +392,115 @@ describe('State editor', function() {
+    editor.addResponse(
+      'TextInput', forms.toRichText('Good job'), 'Two', true, 'Equals',
+      'One');
+    editor.setDefaultOutcome(forms.toRichText('Try again'));
+
+    editor.moveToState('Two');
+    editor.setContent(forms.toRichText('Please write 2 in words.'));
+    editor.setInteraction('TextInput');
+    editor.addResponse(
+      'TextInput', forms.toRichText('Good job'), 'Three', true, 'Equals',
+      'Two');
+    editor.setDefaultOutcome(forms.toRichText('Try again'));
+    editor.addHint('The number 2 in words.');
+    editor.addSolution('TextInput', {
+      correctAnswer: 'Two',
+      explanation: 'The English equivalent of 2'
+    });
+    editor.moveToState('Three');
+    editor.setContent(forms.toRichText('Please write 3 in words.'));
+    editor.setInteraction('TextInput');
+    editor.addResponse(
+      'TextInput', forms.toRichText('Good job'), 'End', true, 'Equals',
+      'Three');
+    editor.setDefaultOutcome(forms.toRichText('Try again'));
+
+    editor.moveToState('End');
+    editor.setInteraction('EndExploration');
+    editor.saveChanges();
+    workflow.publishExploration();
+
+    users.logout();
+
+    // Learner 1 completes the exploration.
+    users.login('user2@statisticsTab.com');
+    libraryPage.get();
+    libraryPage.playExploration(EXPLORATION_TITLE);
+
+    explorationPlayerPage.submitAnswer('TextInput', 'One');
+    explorationPlayerPage.clickThroughToNextCard();
+    explorationPlayerPage.submitAnswer('TextInput', '2');
+    explorationPlayerPage.viewHint();
+    explorationPlayerPage.clickGotItButton();
+    explorationPlayerPage.submitAnswer('TextInput', '3');
+    explorationPlayerPage.viewSolution();
+    explorationPlayerPage.clickGotItButton();
+    explorationPlayerPage.submitAnswer('TextInput', 'Two');
+    explorationPlayerPage.clickThroughToNextCard();
+    explorationPlayerPage.expectExplorationToNotBeOver();
+    explorationPlayerPage.submitAnswer('TextInput', 'Three');
+    explorationPlayerPage.clickThroughToNextCard();
+    explorationPlayerPage.expectExplorationToBeOver();
+
+    users.logout();
+
+    // Learner 2 starts the exploration and immediately quits it.
+    users.login('user3@statisticsTab.com');
+    libraryPage.get();
+    libraryPage.playExploration(EXPLORATION_TITLE);
+
+    explorationPlayerPage.expectExplorationToNotBeOver();
+
+    users.logout();
+
+    // Creator visits the statistics tab.
+    users.login('user1@statisticsTab.com');
+    creatorDashboardPage.get();
+    creatorDashboardPage.navigateToExplorationEditor();
+    editor.navigateToStatsTab();
+
+    // Now, there should be one passerby for this exploration since only learner
+    // 3 quit at the first state.
+    editor.expectNumPassersbyToBe('1');
+
+    users.logout();
+  });
+
   afterEach(function() {
-    general.checkForConsoleErrors([]);
+    general.checkForConsoleErrors([
+      'TypeError: google.visualization.PieChart is not a constructor'
```

Comment:
Let's add a TODO comment here to make sure we understand why this console error appears.

Reply:
Done.